app/services/foreman_ansible/roles_importer.rb

```diff
@@ -1,24 +1,55 @@
 module ForemanAnsible
-  # Service to import roles from the filesystem.
-  # Should be extracted to a gem that both the proxy and Foreman can use.
+  # imports roles from smart proxy
   class RolesImporter
-    class << self
-      def import!
-        list_roles_in_fs.each do |role_name|
-          role = AnsibleRole.new(:name => role_name)
-          next if role.save
-          Rails.logger.debug("Failed to save role #{role_name}: "\
-                            "#{role.errors.full_messages.join(', ')}")
-        end
+    attr_reader :ansible_proxy
+
+    def initialize(proxy = nil)
+      @ansible_proxy = proxy
+    end
+
+    def proxy_api
+      return @proxy_api if @proxy_api
+      find_proxy_api
+    end
+
+    def import!
+      imported = proxy_api.roles.map do |role_name|
+        ::AnsibleRole.find_or_initialize_by(:name => role_name)
+      end
+      detect_changes imported
+    end
+
+    def finish_import(changes)
+      if changes
```

Comment:
Use a guard clause, e.g `return unless changes.present?` to avoid making "if" methods.